### PR TITLE
Add and remove suggested edits on LoginLogoutEvent.

### PR DESCRIPTION
**Pull request by sharvaniharan**
- Suggested Edits tab should be shown only when users are logged in.

### CODE REVIEW

**Comment by sharvaniharan** [comment]
Thank you @joewalsh

File: app/src/main/java/org/wikipedia/navtab/NavTabLayout.java [hunk 1]
```diff
@@ -23,10 +25,20 @@ public NavTabLayout(Context context, AttributeSet attrs, int defStyleAttr) {
         setTabViews();
     }
 
-    private void setTabViews() {
+    public void setTabViews() {
         for (int i = 0; i < NavTab.size(); i++) {
             NavTab navTab = NavTab.of(i);
-            getMenu().add(Menu.NONE, i, i, navTab.text()).setIcon(navTab.icon());
+            if (AccountUtil.isLoggedIn()) {
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/navtab/NavTabLayout.java]: Can we simplify the logic by doing the following:
```
            if (!AccountUtil.isLoggedIn() && NavTab.SUGGESTED_EDITS == navTab) {
                continue;
            }
```

File: app/src/main/java/org/wikipedia/navtab/NavTabLayout.java [hunk 2]
```diff
@@ -23,10 +25,20 @@ public NavTabLayout(Context context, AttributeSet attrs, int defStyleAttr) {
...
+            } else {
+                if (!(NavTab.SUGGESTED_EDITS == navTab)) {
+                    getMenu().add(Menu.NONE, i, i, navTab.text()).setIcon(navTab.icon());
+                }
+            }
         }
     }
+
+    public void clear() {
+        getMenu().clear();
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/navtab/NavTabLayout.java]: Not sure but maybe we can add `getMenu().clear()` to the beginningg of `setTabViews()`.

**Comment by sharvaniharan** [comment]
changes done @cooltey

File: app/src/main/java/org/wikipedia/WikipediaApp.java [hunk 3]
```diff
@@ -381,6 +382,8 @@ public synchronized void resetWikiSite() {
     public void logOut() {
         L.d("Logging out");
         AccountUtil.removeAccount();
+        WikipediaApp.getInstance().getBus().post(new LoginLogoutEvent());
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/WikipediaApp.java]: Not sure but do we really need a `Bus` event to update the Nav tab view? Please see `loginLogoutClick()` in `MainActivity`

File: app/src/main/java/org/wikipedia/login/LoginActivity.java [hunk 4]
```diff
@@ -180,6 +181,7 @@ private void onLoginSuccess() {
         Prefs.setReadingListPagesDeletedIds(Collections.emptySet());
         Prefs.setReadingListsDeletedIds(Collections.emptySet());
         ReadingListSyncAdapter.manualSyncWithForce();
+        WikipediaApp.getInstance().getBus().post(new LoginLogoutEvent());
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/login/LoginActivity.java]: Same question. Maybe you can check my implementation in #632 to see if it works for you.

**Comment by sharvaniharan** [comment]
changes done @cooltey

**Pull request merged**